std/file.d

```diff
@@ -4005,7 +4005,23 @@ unittest
 
 
 /**
-Reads an entire file into an array.
+ * Reads a file line by line and parses the line into a single value or a
+ * $(REF Tuple, std,typecons) of values depending on the length of `Types`.
+ * The lines are parsed using the specified format string. The format string is
+ * passed to $(REF formattedRead, std,format), and therefore must conform to the
+ * format string specification outlined in $(MREF format).
+ *
+ * Params:
+ *     Types = the types that each of the elements in the line should be returned as
+ *     filename = the name of the file to read
+ *     format = the format string to use when reading
+ *
+ * Returns:
+ *     If only one type is passed, then an array of that type. Otherwise, an
+ *     array of $(REF Tuple, std,typecons)s.
+ *
+ * Throws:
+ *     `Exception` if one of the lines is empty
```

Comment:
It also throws if there are trailing characters after the format string. At least, I can see an `enforce` down there.

Also, formattedRead has an enforce about the format string, which might throw. This is not documented as well. :disappointed: